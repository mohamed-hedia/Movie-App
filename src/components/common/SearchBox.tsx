import { ChangeEvent, FC, KeyboardEvent } from "react";
import { useSearchParams } from "react-router-dom";
import { IoSearch } from "react-icons/io5";

const SearchBox: FC = () => {
  const [, setSearchParams] = useSearchParams("");

  const handleSearch = (
    event: KeyboardEvent<HTMLInputElement> & ChangeEvent<HTMLInputElement>
  ) => {
    const inputValue = event.target.value;
    if (event.key === "Enter") {
      const newSearchParams: Record<string, string> = {};
      newSearchParams["search"] = inputValue;
      setSearchParams(newSearchParams, { replace: true });
    }
  };

  return (
    <div className="w-full flex gap-6 items-start justify-center md:mt-3 pr-6">
      <div className="w-9 h-9 mt-2">
        <IoSearch className="w-full h-full" />
      </div>
      <input
        className="w-full bg-transparent py-3 text-xl outline-none focus:border-b-2 caret-orange truncate text-ellipsis"
        type="text"
        placeholder="Search for movies and TV series..."
        onKeyDown={handleSearch}
        required
      />
    </div>
  );
};
export default SearchBox;
