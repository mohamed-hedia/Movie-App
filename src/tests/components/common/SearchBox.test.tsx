import SearchBox from "@/components/common/SearchBox";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";

describe("SearchBox", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Router>
        <SearchBox />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it("has a search input field", () => {
    render(
      <Router>
        <SearchBox />
      </Router>
    );
    const input = screen.getByPlaceholderText(
      "Search for movies and TV series..."
    );
    expect(input).toBeInTheDocument();
  });

  it("has an empty input field on initial render", () => {
    render(
      <Router>
        <SearchBox />
      </Router>
    );
    const input = screen.getByPlaceholderText(
      "Search for movies and TV series..."
    );
    expect(input).toHaveValue("");
  });
  
  it("updates input field when text is entered", async() => {
    render(
      <Router>
        <SearchBox />
      </Router>
    );
    const input = screen.getByPlaceholderText(
      "Search for movies and TV series..."
    );
    const user = userEvent.setup();
    await user.type(input, "test");
    expect(input).toHaveValue("test");
  });
});
