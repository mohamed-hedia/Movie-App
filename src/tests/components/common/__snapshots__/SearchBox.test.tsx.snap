// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBox > renders correctly 1`] = `
<div>
  <div
    class="w-full flex gap-6 items-start justify-center md:mt-3 pr-6"
  >
    <div
      class="w-9 h-9 mt-2"
    >
      <svg
        class="w-full h-full"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M456.69 421.39 362.6 327.3a173.81 173.81 0 0 0 34.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 0 0 327.3 362.6l94.09 94.09a25 25 0 0 0 35.3-35.3zM97.92 222.72a124.8 124.8 0 1 1 124.8 124.8 124.95 124.95 0 0 1-124.8-124.8z"
        />
      </svg>
    </div>
    <input
      class="w-full bg-transparent py-3 text-xl outline-none focus:border-b-2 caret-orange truncate text-ellipsis"
      placeholder="Search for movies and TV series..."
      required=""
      type="text"
    />
  </div>
</div>
`;
